packages/modules/currency/src/models/currency.ts

```diff
@@ -1,50 +1,10 @@
-import { BigNumberRawValue } from "@medusajs/types"
-import {
-  BigNumber,
-  MikroOrmBigNumberProperty,
-  Searchable,
-} from "@medusajs/utils"
-import { Entity, PrimaryKey, Property } from "@mikro-orm/core"
-
-@Entity({ tableName: "currency" })
-class Currency {
-  @Searchable()
-  @PrimaryKey({ columnType: "text" })
-  code!: string
-
-  @Property({ columnType: "text" })
-  symbol: string
-
-  @Property({ columnType: "text" })
-  symbol_native: string
-
-  @Searchable()
-  @Property({ columnType: "text" })
-  name: string
-
-  @Property({ columnType: "int", default: 0 })
-  decimal_digits: number
-
-  @MikroOrmBigNumberProperty({ default: 0 })
-  rounding: BigNumber | number
-
-  @Property({ columnType: "jsonb" })
-  raw_rounding: BigNumberRawValue
-
-  @Property({
-    onCreate: () => new Date(),
-    columnType: "timestamptz",
-    defaultRaw: "now()",
-  })
-  created_at: Date
-
-  @Property({
-    onCreate: () => new Date(),
-    onUpdate: () => new Date(),
-    columnType: "timestamptz",
-    defaultRaw: "now()",
-  })
-  updated_at: Date
-}
-
-export default Currency
+import { model } from "@medusajs/utils"
+
+export default model.define("currency", {
+  code: model.id({ generateId: false }),
```

Comment:
is this the same as `.text().primaryKey()`?

Reply:
^^ I don't think we should add `generateId` to the `id` property. That's a confusing API imo.

Instead, we should encourage the approach @carlos-r-l-rodrigues highlights here for such cases.

Reply:
I came here to same the same thing as @carlos-r-l-rodrigues 😄 i think using `id` is very confusing. `primaryKey` or just `primary` should be good

Reply:
updated 🤞🏻